fluent/bin/utils.py

```
@@ -0,0 +1,190 @@
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+"""
+This file contains utility functions to use with nupic.fluent experiments.
+"""
+
+import csv
+import os
+
+from collections import Counter
+
+
+
+exclusions = ('!', '.', ':', ',', '"', '\'', '\n', '?')
+
+
+################################################################################
+# Text processing functions:
+
+def tokenize(string, ignoreCommon=False):
+  """
+  Tokenize the string into a list of strings, leaving out all chars in the
+  exclusions list.
+  """
+  if not isinstance(string, str):
+    raise ValueError("Must input a single string object to tokenize.")
+  line = "".join([c for c in string if c not in exclusions])
+  strings = line.split(" ")
+  if ignoreCommon:
+  	common = getFrequentWords()
+  	strings = [s for s in strings if s not in common]
+  return strings
+
+
+def getFrequentWords(n=200):
+	"""Returns the n most frequent English words/lemmas."""
+	if n>1000:
+		raise ValueError("Max number of words is 1000.")
+	datafile = os.path.abspath(os.path.join(os.path.dirname(__file__), '../..', 'data/etc/word_frequencies.txt'))
+	try:
+		with open(datafile) as f:
+			lst = []
+			for i, line in enumerate(f):
+				if i==0: continue
+				if i>99:
+					lst.append(line.split(' ')[3])
+				elif i>9:
+					lst.append(line.split(' ')[4])
+				else:
+					lst.append(line.split(' ')[5])
+				if i==n: break
+			return lst
+	except IOError:
+		print ("Cannot find the word frequencies data file.")
```

More generic error message. Raise an exception instead of print.